### DISCUSSION TITLE
Windows installation instructions

### DISCUSSION THREAD

**Discussion by mbassano** [aedocw/epub2tts-edge#41] [Q&A]
I'd appreciate an installation instruction for Windows, since I have no experience with WSL and/or Ubuntu.

**Comment by aedocw** [comment]
If you could try these steps and report back, if they work properly I'll be happy to update the README.

1. [Install chocolaty](https://chocolatey.org/install)
2. Install ffmpeg with the command `choco install ffmpeg`, make sure you are in an elevated powershell session.
3. Install python 3.11 with the command `choco install python311`
4. Install git with the command `choco install git`.
5. Decide where you want your epub2tts-edge project to live, Documents is a common place. Once you've found a directory you're happy with, clone the project with `git clone https://github.com/aedocw/epub2tts-edge` and `cd epub2tts-edge` so you're now in your working directory.
6. There are probably a few different ways you can go here, I personally opted for a venv to keep everything organized. Create a venv with the command `python -m venv .venv`
7. Activate the venv, on windows the command is slightly different as you issue `.venv\scripts\activate`
8. Install epub2tts-edge along with the requirements with the `command pip install .`
9. If all goes well, you should be able to call epub2tts-edge from within your venv and update it from this directory going forward. To update, activate the virtual environment then use `git pull` and then `pip install . --upgrade`

**Reply by mbassano** [reply]
Hi Chris,
Thanks for the instruction. It failed at step 5 however:

PS D:\Windows Folders\Michael\Documents> git clone https://github.com/aedocw/epub2tts-edge
git : The term 'git' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the s
pelling of the name, or if a path was included, verify that the path is correct and try again.
At line:1 char:1
+ git clone https://github.com/aedocw/epub2tts-edge
+ ~~~
    + CategoryInfo          : ObjectNotFound: (git:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException

**Reply by aedocw** [reply]
What happened with step 4? `choco install git` should have installed git for you. If that failed, probably installing ffmpeg failed too? Take a close look at choco install steps and output, it's also possible you might need to open a new terminal after installing choco (I can't remember the install instructions for that), or maybe even have to reboot before choco works completely. That's going to be an important part to get right though :)

**Reply by mbassano** [reply]
You're right it said I would need to reopen PowerShell after git installation. I had missed that. In step 8 "command" should not be part of the actual command. Everything should be installed now. Now trying to figure out how to use it.... ;-)

**Reply by aedocw** [reply]
Excellent! Hope it works from here for you, and have fun!

**Reply by mbassano** [reply]
Well, my first attempt failed, unfortunately:

epub2tts-edge "E:\eBooks\Fiction\Star Trek\Star Trek - Deep Space Nine\Star Trek_ Deep Space Nine - 001 - Emissary - J. M. Dillard & Michael Piller & Rick Berman.txt" --cover "E:\eBooks\Fiction\Star Trek\Star Trek - Deep Space Nine\Star Trek_ Deep Space Nine - 001 - Emissary - J. M. Dillard & Michael Piller & Rick Berman.png"
Namespace(sourcefile='E:\\eBooks\\Fiction\\Star Trek\\Star Trek - Deep Space Nine\\Star Trek_ Deep Space Nine - 001 - Emissary - J. M. Dillard & Michael Piller & Rick Berman.txt', speaker='en-US-AndrewNeural', cover='E:\\eBooks\\Fiction\\Star Trek\\Star Trek - Deep Space Nine\\Star Trek_ Deep Space Nine - 001 - Emissary - J. M. Dillard & Michael Piller & Rick Berman.png', sentencepause=1200, paragraphpause=1200)
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "D:\Windows Folders\Michael\Documents\epub2tts-edge\.venv\scripts\epub2tts-edge.exe\__main__.py", line 7, in <module>
  File "D:\Windows Folders\Michael\Documents\epub2tts-edge\.venv\Lib\site-packages\epub2tts_edge\epub2tts_edge.py", line 438, in main
    book_contents, book_title, book_author, chapter_titles = get_book(args.sourcefile)
                                                             ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Windows Folders\Michael\Documents\epub2tts-edge\.venv\Lib\site-packages\epub2tts_edge\epub2tts_edge.py", line 166, in get_book
    for line in file:
  File "<frozen codecs>", line 322, in decode
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x92 in position 151: invalid start byte


I'll try again with a different ebook another time. Got to stop now.

**Comment by aedocw** [comment]
Almost sounds like it's not a plain-text file. Were you able to open that file with notepad or some other plain text editor to confirm that? I like to edit the text file before sending it to be read anyway, usually removing things like table of contents, etc.

Just to confirm things are mostly working, try sending any really short text file through epub2tts-edge to confirm it can at least create the audio.
